File: app/src/main/java/com/example/dishora/models/ChatMessageDto.java
package com.example.dishora.models;

import com.google.gson.annotations.SerializedName;
import java.util.Date;

public class ChatMessageDto {
    @SerializedName("messageId")
    private long messageId;
    @SerializedName("senderId")
    private long senderId;
    @SerializedName("messageText")
    private String messageText;
    @SerializedName("sentAt")
    private Date sentAt;

    // Add getters for all fields
    public long getMessageId() { return messageId; }
    public long getSenderId() { return senderId; }
    public String getMessageText() { return messageText; }
    public Date getSentAt() { return sentAt; }
}

File: DTO/ChatMessageDto.cs
namespace Dishora.DTO
{
    public class ChatMessageDto
    {
        public long message_id { get; set; }
        public long sender_id { get; set; }
        public string? sender_role { get; set; }
        public string? sender_name { get; set; } // "Jomar" or "Lola's Lutong Bahay"
        public string? message_text { get; set; }
        public string? image_url { get; set; } // <--- ADD THIS
        public DateTime sent_at { get; set; }
    }
}
